Content.Client/SubFloor/SubFloorHideSystem.cs

```diff
@@ -67,8 +67,12 @@ private void OnAppearanceChanged(EntityUid uid, SubFloorHideComponent component,
         // allows a t-ray to show wires/pipes above carpets/puddles
         if (scannerRevealed)
         {
-            component.OriginalDrawDepth ??= args.Sprite.DrawDepth;
-            args.Sprite.DrawDepth = (int) Shared.DrawDepth.DrawDepth.FloorObjects + 1;
+            if (component.OriginalDrawDepth is not null)
+                return;
+            component.OriginalDrawDepth = args.Sprite.DrawDepth;
+            var drawDepthDifference =
+                Math.Abs((int)Shared.DrawDepth.DrawDepth.ThickPipe - (int)Shared.DrawDepth.DrawDepth.FloorObjects);
+            args.Sprite.DrawDepth += drawDepthDifference + 1;
         }
```

Comment:
Abs is not what you want, if it's already higher it's just going to amplify the difference even further.